core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,94 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JavaGrepLambdaImp extends JavaGrepImp {
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("USAGE: JavaGrep regex rootPath outFile");
+    }
+
+    //creating JavaGrepLambdaImp instead of JavaGrepImp
+    //JavaGrepLambdaImp inherits all methods except two override methods in this class.
+    JavaGrepLambdaImp javaGrepLambdaImp = new JavaGrepLambdaImp();
+    javaGrepLambdaImp.setRegex(args[0]);
+    javaGrepLambdaImp.setRootPath(args[1]);
+    javaGrepLambdaImp.setOutFile(args[2]);
+
+    try {
+      //calling parent method, but it will call override method (in this class)
+      javaGrepLambdaImp.process();
+    } catch (Exception ex) {
+      ex.printStackTrace();
+    }
+
+  }
+
+  @Override
+  public void process() throws IOException {
+    File rootDir = new File(this.getOutFile());
```

Comment:
where is getOutFile method in this class?

Reply:
The class JavaGrepLambdaImp inherits all the mentioned methods from JavaGrepImp.